os/src/main/java/io/smallrye/common/os/GetAllProcessesInfoAction.java

```diff
@@ -0,0 +1,12 @@
+package io.smallrye.common.os;
+
+import java.security.PrivilegedAction;
+import java.util.List;
+
+public class GetAllProcessesInfoAction implements PrivilegedAction<List<ProcessInfo>> {
+
+    @Override
+    public List<ProcessInfo> run() {
+        throw new UnsupportedOperationException("Listing all processes is not supported in JDK 8");
```

Comment:
On Linux, you could open `/proc` and iterate each directory, something like this:

```java
        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get("/proc"))) {
            for (Path procPath : stream) {
                String name = procPath.getFileName().toString();
                if (name.matches("\\d+")) {
                    long pid = Long.parseLong(name);
                    try (BufferedReader reader = Files.newBufferedReader(procPath.resolve("cmdline"), StandardCharsets.UTF_8)) {
                        String line = reader.readLine();
                        int idx = line.indexOf(0);
                        String cmdLine = idx == -1 ? line : line.substring(0, idx);
                        // new ProcessInfo(pid, cmdLine);
                    }
                }
            }
        }
```

I don't know what the Mac or Windows equivalents are, but it might be worth looking into (even if it ends up just being a parse of `ps` or something).

Reply:
@dmlloyd can you fix CI to also build in Windows and MacOS? I am developing under Fedora 33 so I don't know if the fix would work in other OSes

Reply:
I have a PR in testing now, just have to get Windows right and then I'll merge it.  Then you can rebase and you'll have it.

Reply:
Great stuff! Thanks guys! Now I won't have to rely on on reflection to do this in Quarkus :)

Reply:
It turns out we have test failures on Windows.  I guess adding CI configurations was a pretty good idea. :)

Reply:
OK that's merged.  Go ahead and rebase and you should be tested on all three platforms.

Reply:
Awesome! Bonus points if it's built with different JDK versions too :)

Reply:
This is a SmallRye MR-JAR project, so it always builds with Java 11 and tests with 8, 9, 10, and 11.